### PR TITLE
[Runtime] Let CF provide us with state, avoiding runtime lookup. rdar://111104451

### PR DESCRIPTION
**Pull request by Catfish-Man**
We can avoid a bunch of calls to `objc_lookUpClass()` if we have CF pass in the information we need, which it already knows.

Fixes rdar://111104451

### CODE REVIEW

File: stdlib/public/stubs/FoundationHelpers.mm [hunk 1]
```diff
@@ -31,36 +31,105 @@
...
+  return bridgingState->NSErrorClass;
+}
+
+SWIFT_CC(swift) SWIFT_RUNTIME_STDLIB_SPI
+bool
+swift_stdlib_connectNSBaseClasses() {
+  initializeBridgingFunctions();
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+  Class ourArray = objc_lookUpClass("__SwiftNativeNSArrayBase");
```
Comment (Catfish-Man) [inline stdlib/public/stubs/FoundationHelpers.mm:105]: Note to self: make sure to change this to not re-do the work if CF already did it

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
@swift-ci please benchmark

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
ok! Tests green. Mike noticed a few things to tweak, so retesting with those now.

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
@swift-ci please test

File: stdlib/public/stubs/SwiftNativeNSXXXBase.mm.gyb [hunk 2]
```diff
@@ -87,22 +87,6 @@ swift_stdlib_NSObject_isEqual(id lhs,
   return (lhs == rhs) || [lhs isEqual:rhs];
 }
 
-SWIFT_CC(swift) SWIFT_RUNTIME_STDLIB_SPI
```
Comment (al45tair) [inline stdlib/public/stubs/SwiftNativeNSXXXBase.mm.gyb:90]: Oh, getting rid of that is really nice :-) That halves the number of class lookup-by-string calls we're doing.

**Review by al45tair — APPROVED** [review]
Like it :-)

**Comment by al45tair** [comment]
BTW, I tweaked the description of the PR a little; the only thing missing is a Radar number :-) Really nice change.

**Pull request merged**